core/src/ast_builder/index.rs

```diff
@@ -0,0 +1,94 @@
+#![cfg(feature = "index")]
+
+use crate::{
+    ast::{Expr, ObjectName, OrderByExpr, Statement},
+    result::Result,
+};
+
+#[derive(Clone)]
+pub struct CreateIndexNode {
+    name: String,
+    table_name: String,
+    column: String,
+    asc: bool,
```

Comment:
https://github.com/gluesql/gluesql/blob/0d2ecc5f2b00a006b1e84417de21c3b8388d3859/core/src/ast/mod.rs#L82-L87

type of field `column` in `Statement::CreateIndex` is `OrderByExpr`

https://github.com/gluesql/gluesql/blob/0d2ecc5f2b00a006b1e84417de21c3b8388d3859/core/src/ast/query.rs#L102-L106

Then, here what we need to do is adding a new node - `OrderByExprNode`.
One expected behavior of `OrderByExprNode` is taking `"id + name` or `col1 || col2 DESC` like strings and converts into `ast::OrderByExpr`.